### DISCUSSION TITLE
Is it faster to pre-upload/sync/clone folders with many small files (photos)?

### DISCUSSION THREAD

**Discussion by ngoonee** [abraunegg/onedrive#2199] [Q&A]
I have 161 GB of mostly photos (some videos), roughly 115 thousand files in total. Was previously kept in sync between multiple machines using syncthing, and trying to use onedrive on multiple PCs instead.

So say I have PCs A, B, and C all of whom contain a mostly complete copy currently. Let's treat A as the 'main' machine.

Based on https://github.com/abraunegg/onedrive/discussions/2186 it would seem that once the initial sync to onedrive is complete from A, I can just add the right config/folders to B and C. That initial sync takes ages though, would it be safe to manually upload on the web interface (or perhaps use rclone) the entire contents of the folder, and then just run a --synchronize? Or is it safer to just let onedrive 'do its thing' slowly?

**Comment by abraunegg** [comment]
@ngoonee 
Your probably looking at the problem the wrong way. Remember - OneDrive is *not* designed to be a backup service.

I would investigate doing the following for important data such as photos:

* local pc, running linux, using zfs raidz2 or raidz3 for your local data 'on prem' - local to you (at home for example)
* configure a zfs volume, with say copies = 2, so you have multiple ways to recover any lost file fragment. Between raidz2/3 + copies =2 .. you have data integrity and resiliency matching your data importance.

From this same 'server' - configure the onedrive client to sync this specific zfs volume containing all your data to OneDrive as an upload only, with no remote delete.

This then puts your local data to OneDrive as a backup ...... but just remember OneDrive should not be used as a backup service.

As a tertiary mechanism, I would then use Backblaze B2 or AWS Glacier or Google Storage (not Google Drive) and a secondary location.

I would NOT , under any circumstance, configure 3 separate systems and upload to the same OneDrive account data, regardless of configuration or desire. This approach is asking for issues and problem and heartache.

Adsitionally - do NOT use rclone and 'onedrive' together / at the same time / on different systems. Use one or the other, not both. This is due to how the OneDrive API has beem designed - that is, a major assumption that only 1 client is bsing used - by using 2 different ones, all the /delta calls and reaponses could cause you issues and data loss.

**Reply by ngoonee** [reply]
Thanks @abraunegg - some clarifications on the last 2 paragraphs:-
1. 'configure 3 separate systems and upload' is how onedrive is commonly used on Windows PCs, is it not? Seems to me a quite primary purpose is to keep files in sync across multiple PCs for device-agnostic use. Obviously if there are conflicting changes this would need to be manually fixed as there's no possibility of automating that, but is data loss/corruption a worry in this scenario?
2. My idea is not to use them together/at the same time. I'm asking about using rclone one-way to initially populate. So let's say I have a single folder PHOTOS with many subfolders in my ~/OneDrive/PHOTOS. I can either
a) Setup sync using the onedrive app and wait for it to sync everything (this seems slow, which makes sense!)
b) rclone the entirety of PHOTOS to onedrive:PHOTOS, and then setup the sync (after rclone is complete). So rclone only used for the initial data population. My idea is that this would work similarly to when you do a resync on onedrive, where it rebuilds the cache  by comparing all files (but presumably since only dealing with descriptors/metadata and not having to transfer the files themselves, this would be faster)

Is b) a legitimate option, or would this break everything?

**Comment by abraunegg** [comment]
@ngoonee 
> 'configure 3 separate systems and upload' is how onedrive is commonly used on Windows PCs, is it not?

On Windows, Microsoft assumes that *everything* stored in the Windows OneDrive folder is what is put online, what is stored online is stored back on Windows. There is zero client side filtering.

On Linux, you can do client side filtering + a whole host of other things. 

Just because you can do this with Windows, does not mean you can do the exact same on Linux ... yes it will work, but it will also cause a lot of heartache + issues around who is 'data master' - what PC is the primary data set, who uploads what / this is missing / that is missing  - delete ..............

>  Obviously if there are conflicting changes this would need to be manually fixed as there's no possibility of automating that, but is data loss/corruption a worry in this scenario?

Absolutely a major data-loss scenario if you do things incorrectly.

> Is b) a legitimate option, or would this break everything?

Do not use 'rclone' and 'onedrive' together with the same account. Use one or the other - not both. Take this as a major warning.

You should seriously consider your local data integrity and position. You need to investigate ZFS and raidz2/3 and ZFS Volumes with copies=2 .. you have data integrity and resiliency matching your data importance. Do not have 2 or 3 separate systems (unless these are at totally separate locations ..... even then however, ZFS should be used to store the data) trying to sync your data.
